packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -0,0 +1,422 @@
+import { getRetryProvider } from "@uma/common";
+import { MulticallMakerDaoEthers } from "@uma/contracts-node";
+import {
+  aggregateTransactionsAndCall,
+  delay,
+  NetworkerInterface,
+  TransactionDataDecoder,
+} from "@uma/financial-templates-lib";
+import { getContractInstanceWithProvider } from "../utils/contracts";
+
+import type { Provider } from "@ethersproject/abstract-provider";
+import request from "graphql-request";
+
+import { ethers } from "ethers";
+
+import Web3 from "web3";
+import { ProposePriceEvent } from "@uma/contracts-node/dist/packages/contracts-node/typechain/core/ethers/OptimisticOracleV2";
+
+export { Logger } from "@uma/financial-templates-lib";
+export { getContractInstanceWithProvider } from "../utils/contracts";
+
+const { Datastore } = require("@google-cloud/datastore");
+const datastore = new Datastore();
+
+// Helper function to sleep for a given duration
+const sleep = (ms: number) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+
+export interface BotModes {
+  transactionsProposedEnabled: boolean;
+}
+
+export interface BlockRange {
+  start: number;
+  end: number;
+}
+
+export interface MonitoringParams {
+  binaryAdapterAddress: string;
+  ctfAdapterAddress: string;
+  graphqlEndpoint: string;
+  apiEndpoint: string;
+  provider: Provider;
+  chainId: number;
+  blockRange: BlockRange;
+  pollingDelay: number;
+  botModes: BotModes;
+}
+
+interface PolymarketMarket {
+  resolvedBy: string;
+  questionID: string;
+  createdAt: string;
+  question: string;
+  outcomes: [string, string];
+  outcomePrices: [string, string];
+  liquidityNum: number;
+  volumeNum: number;
+  clobTokenIds: [string, string];
+}
+
+interface PolymarketMarketWithAncillaryData extends PolymarketMarket {
+  ancillaryData: string;
+}
+
+export interface PolymarketWithEventData extends PolymarketMarketWithAncillaryData {
+  txHash: string;
+  requester: string;
+  proposer: string;
+  timestamp: number;
+  expirationTimestamp: number;
+  eventTimestamp: number;
+  eventBlockNumber: number;
+  eventIndex: number;
+  proposalTimestamp: number;
+  identifier: string;
+  ancillaryData: string;
+  proposedPrice: string;
+}
+
+interface PolymarketOrderBook {
+  market: string;
+  asset_id: string;
+  bids: { price: string; size: string }[];
+  asks: { price: string; size: string }[];
+  hash: string;
+}
+
+export type OrderBookSide = { price: number; size: number }[];
+export interface MarketOrderbooks {
+  orderBooks: [{ bids: OrderBookSide; asks: OrderBookSide }, { bids: OrderBookSide; asks: OrderBookSide }];
+}
+
+export interface StoredNotifiedProposal {
+  txHash: string;
+  question: string;
+  proposedPrice: string;
+  notificationTimestamp: number;
+}
+
+// Helper function to process markets in chunks
```

Comment:
Just so I understand, is this to avoid some sort of API rate limit? Or some other purpose?

Reply:
Yes, otherwise the api timeouts, specially during my initial tests when I was testing with more markets with the previous strategy. Maybe it's not needed anymore, I can simplify this so it's more easy to reason about.